### DISCUSSION TITLE
Set a "divert" variable from gdscript?

### DISCUSSION THREAD

**Discussion by mlinden87** [ephread/inkgd#62] [Q&A] [answered]
I'm using Ink to setup many short dialogues in my game, where a knot represents a separate dialogue. In the game, when interacting with certain NPCs the player gets offered a couple dialogues to choose from. Choosing one diverts to the chosen ink knot to start the dialogue. Everything works, except for the actual divert.

Very simplified, in Ink I set up variables for the knots to divert to.

**In Ink:**
```
VAR option_1_divert = -> my_knot_x
VAR option_2_divert = -> my_knot_y

What do you want to talk about?
*[Option_1] -> option_1_divert
*[Option_2] -> option_2_divert
```

On the Godot side of things, I have a function that creates a random knot that exists in the ink story based on a couple criteria. This should then be passed to the ink variable `VAR option_1_divert` to divert to the random knot when choosing that option. Problem is, this is a string, so can't pass it back into the ink variable, as that expects a divert type.

**In Godot:**
```GDScript
var random_knot_1: String = create_random_knot() #returns e.g. awesome_knot_z

_ink_player.set_variable("option_1_divert", random_knot_1)
```

Is there a way to make this work? Found an old discussion for Unity, that offered a way by changing the Path class, but how about Godot?

**Comment by ephread** [comment, marked as answer]
> Is there a way to make this work? Found an old discussion for Unity, that offered a way by changing the Path class, but how about Godot?

Yes,`Path` exists in _inkgd_ as well, it's in [`ink_path.gd`](https://github.com/ephread/inkgd/blob/ff64af3ceaaf165f2afe900faf6c310cc6443fe8/addons/inkgd/runtime/ink_path.gd) (to prevent collisions with the native `Path` type).

```gdscript
var InkPath := load("res://addons/inkgd/runtime/ink_path.gd") as GDScript

var random_knot_1 = InkPath.new_with_components_string("awesome_knot_z")

_ink_player.set_variable("option_1_divert", random_knot_1)
```

Note that creating paths that point to gathers outside of the runtime is tricky because gathers are usually one container deeper than expected. For instance, this divert:

```ink
VAR divert = -> knot.stitch.gather
```

uses the following path:

```gdscript
var divertPath = _ink_player.get_variable("divert") as InkPath
var path = divertPath.components_string # equals "knot.stitch.0.gather"
```

See [here](https://github.com/ephread/inkgd/blob/ff64af3ceaaf165f2afe900faf6c310cc6443fe8/test/integration/player/test_variables_setter_getter.gd#L49) and [here](https://github.com/ephread/inkgd/blob/ff64af3ceaaf165f2afe900faf6c310cc6443fe8/addons/inkgd/runtime/ink_path.gd#L245). Let me know if that helps you!

**Reply by mlinden87** [reply]
This is perfect! Works like a charm. Thank you so much!
